File: lib/models/strangers_meet_request.dart
class StrangersMeetRequest {
  final String id;
  final String? userId;
  final String subject;
  final String tagline;
  final DateTime eventDateTime;
  final int numberOfPersons;
  final double chargesPerHead;
  final int slotsFilled;
  final String status;
  final double? paymentAmount;
  final String paymentStatus;
  final String? adminNotes;
  final String? ticketId;
  final String? ticketUrl;
  final DateTime? createdAt;
  final String? mobileNumber;
  final String? alternateMobileNumber;

  // v2: Structured bank/UPI details provided at creation
  final String? bankName;
  final String? accountNumber;
  final String? accountHolderName;
  final String? ifscCode;
  final String? upiId;
  final String? upiNumber;

  // Admin-calculated platform charge per seat (= paymentAmount / numberOfPersons)
  final double? platformChargePerSeat;

  // Settlement info
  final String settlementStatus;
  final String? bankDetails; // legacy
  final String? settlementTransactionId;
  final double? settlementAmount;
  final DateTime? settlementDate;
  final String? settlementMethod;

  // Lifecycle fields
  final DateTime? startedAt;
  final String? startedBy;
  final double? durationHours;
  final DateTime? expectedEndAt;
  final DateTime? endedAt;
  final String? endedConfirmedBy;
  final DateTime? endedConfirmedAt;
  final DateTime? adminConfirmedEndedAt;
  final String? adminConfirmedBy;
  final bool settlementOverdue;

  // Dynamic computed values from server
  final int joinedCount;
  final int paymentCount;
  final int remainingCount;

  final Map<String, dynamic>? user;
  final Map<String, dynamic>? venue;
  final List<dynamic>? joiners;

  StrangersMeetRequest({
    required this.id,
    this.userId,
    required this.subject,
    required this.tagline,
    required this.eventDateTime,
    required this.numberOfPersons,
    required this.chargesPerHead,
    required this.slotsFilled,
    required this.status,
    this.paymentAmount,
    required this.paymentStatus,
    this.adminNotes,
    this.ticketId,
    this.ticketUrl,
    this.createdAt,
    this.mobileNumber,
    this.alternateMobileNumber,
    this.bankName,
    this.accountNumber,
    this.accountHolderName,
    this.ifscCode,
    this.upiId,
    this.upiNumber,
    this.platformChargePerSeat,
    required this.settlementStatus,
    this.bankDetails,
    this.settlementTransactionId,
    this.settlementAmount,
    this.settlementDate,
    this.settlementMethod,
    this.startedAt,
    this.startedBy,
    this.durationHours,
    this.expectedEndAt,
    this.endedAt,
    this.endedConfirmedBy,
    this.endedConfirmedAt,
    this.adminConfirmedEndedAt,
    this.adminConfirmedBy,
    this.settlementOverdue = false,
    this.joinedCount = 0,
    this.paymentCount = 0,
    this.remainingCount = 0,
    this.user,
    this.venue,
    this.joiners,
  });

  factory StrangersMeetRequest.fromJson(Map<dynamic, dynamic> json) {
    final eventDateRaw = json['eventDateTime'] ?? json['event_date_time'] ?? json['bookingDate'] ?? json['partyDate'] ?? json['eventStartAt'] ?? json['date'];
    final timeRaw = json['startTime'] ?? json['partyTime'] ?? json['time'];
    final numPersonsRaw = json['numberOfPersons'] ??
        json['number_of_persons'] ??
        json['totalSeats'] ??
        json['maxPersons'] ??
        json['seats'] ??
        json['maxSeats'] ??
        json['numberOfSeats'] ??
        json['numberOfGuests'] ??
        json['persons'] ??
        (json['plan'] is Map ? json['plan']['numberOfPersons'] ?? json['plan']['number_of_persons'] ?? json['plan']['maxPersons'] ?? json['plan']['totalSeats'] : null) ??
        (json['request'] is Map ? json['request']['numberOfPersons'] ?? json['request']['number_of_persons'] ?? json['request']['totalSeats'] : null) ??
        (json['data'] is Map ? json['data']['numberOfPersons'] ?? json['data']['totalSeats'] : null);
    final chargesPerHeadRaw = json['chargesPerHead'] ??
        json['charges_per_head'] ??
        json['hostChargesPerHead'] ??
        json['charges'] ??
        json['entryFee'] ??
        json['fee'] ??
        (json['plan'] is Map ? json['plan']['chargesPerHead'] ?? json['plan']['charges_per_head'] : null) ??
        (json['request'] is Map ? json['request']['chargesPerHead'] ?? json['request']['charges_per_head'] : null) ??
        (json['metadata'] is Map ? json['metadata']['chargesPerHead'] ?? json['metadata']['charges_per_head'] : null) ??
        (json['data'] is Map ? json['data']['chargesPerHead'] ?? json['data']['charges_per_head'] : null);
    final slotsFilledRaw = json['paymentCount'] ?? json['payment_count'] ?? json['slotsFilled'] ?? json['slots_filled'] ?? json['joinedCount'] ?? json['joined_count'];
    final paymentAmountRaw = json['paymentAmount'] ??
        json['payment_amount'] ??
        json['adminPaymentAmount'] ??
        json['admin_payment_amount'] ??
        json['amount'] ??
        json['totalAmount'] ??
        json['depositAmount'] ??
        (json['plan'] is Map ? json['plan']['paymentAmount'] ?? json['plan']['payment_amount'] ?? json['plan']['adminPaymentAmount'] : null) ??
        (json['request'] is Map ? json['request']['paymentAmount'] ?? json['request']['payment_amount'] ?? json['request']['adminPaymentAmount'] : null) ??
        (json['metadata'] is Map ? json['metadata']['paymentAmount'] ?? json['metadata']['payment_amount'] ?? json['metadata']['adminPaymentAmount'] : null) ??
        (json['data'] is Map ? json['data']['paymentAmount'] ?? json['data']['payment_amount'] ?? json['data']['adminPaymentAmount'] : null);
    final paymentStatusRaw = json['paymentStatus'] ?? json['payment_status'];
    final platformChargeRaw = json['platformChargePerSeat'] ?? json['platform_charge_per_seat'];

    DateTime parsedDate = DateTime.now();
    if (eventDateRaw != null) {
      if (eventDateRaw is DateTime) {
        parsedDate = eventDateRaw.toLocal();
      } else {
        parsedDate = DateTime.tryParse(eventDateRaw.toString())?.toLocal() ?? DateTime.now();
      }
    }
    if (timeRaw != null && timeRaw.toString().trim().isNotEmpty) {
      final cleanTime = timeRaw.toString().toUpperCase().trim();
      final isPm = cleanTime.contains('PM');
      final isAm = cleanTime.contains('AM');
      final timeOnly = cleanTime.replaceAll('AM', '').replaceAll('PM', '').trim();
      final parts = timeOnly.split(':');
      if (parts.isNotEmpty) {
        int? h = int.tryParse(parts[0].trim());
        int m = parts.length > 1 ? (int.tryParse(parts[1].trim()) ?? 0) : 0;
        if (h != null) {
          if (isPm && h < 12) h += 12;
          if (isAm && h == 12) h = 0;
          parsedDate = DateTime(parsedDate.year, parsedDate.month, parsedDate.day, h, m);
        }
      }
    } else if (parsedDate.hour == 5 && parsedDate.minute == 30 && eventDateRaw.toString().endsWith('Z')) {
      parsedDate = DateTime(parsedDate.year, parsedDate.month, parsedDate.day, 0, 0);
    }

    final int numPersons = int.tryParse(numPersonsRaw?.toString() ?? '') ?? 2;
    final double charges = chargesPerHeadRaw != null
        ? (chargesPerHeadRaw is num
            ? chargesPerHeadRaw.toDouble()
            : (double.tryParse(chargesPerHeadRaw.toString()) ?? 0.0))
        : 0.0;
    final double? payAmount = paymentAmountRaw != null
        ? (paymentAmountRaw is num
            ? paymentAmountRaw.toDouble()
            : double.tryParse(paymentAmountRaw.toString()))
        : null;

    final userMap = json['user'] is Map 
        ? Map<String, dynamic>.from(json['user']) 
        : (json['host'] is Map
            ? Map<String, dynamic>.from(json['host'])
            : (json['creator'] is Map
                ? Map<String, dynamic>.from(json['creator'])
                : (json['plan'] is Map && json['plan']['user'] is Map
                    ? Map<String, dynamic>.from(json['plan']['user'])
                    : (json['firstName'] != null || json['first_name'] != null
                        ? Map<String, dynamic>.from(json)
                        : null))));
    final venueMap = json['venue'] is Map ? Map<String, dynamic>.from(json['venue']) : null;

    return StrangersMeetRequest(
      id: (json['id'] ?? json['bookingId'] ?? '').toString(),
      userId: json['userId'] ?? json['user_id'] ?? userMap?['id']?.toString(),
      subject: (json['subject'] ?? json['tablePackage'] ?? 'Strangers Meetup').toString(),
      tagline: (json['tagline'] ?? '').toString(),
      eventDateTime: parsedDate,
      numberOfPersons: numPersons,
      chargesPerHead: charges,
      slotsFilled: int.tryParse(slotsFilledRaw?.toString() ?? '') ?? 0,
      status: (json['status'] ?? 'pending').toString(),
      paymentAmount: payAmount,
      paymentStatus: (paymentStatusRaw ?? 'unpaid').toString(),
      adminNotes: json['adminNotes']?.toString() ?? json['admin_notes']?.toString(),
      ticketId: json['ticketId']?.toString() ?? json['ticket_id']?.toString() ?? json['ticketCode']?.toString(),
      ticketUrl: json['ticketUrl']?.toString() ?? json['ticket_url']?.toString(),
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString())?.toLocal()
          : (json['created_at'] != null ? DateTime.tryParse(json['created_at'].toString())?.toLocal() : null),
      mobileNumber: json['mobileNumber']?.toString() ?? json['mobile_number']?.toString(),
      alternateMobileNumber: json['alternateMobileNumber']?.toString() ?? json['alternate_mobile_number']?.toString(),
      bankName: json['bankName']?.toString() ?? json['bank_name']?.toString(),
      accountNumber: json['accountNumber']?.toString() ?? json['account_number']?.toString(),
      accountHolderName: json['accountHolderName']?.toString() ?? json['account_holder_name']?.toString(),
      ifscCode: json['ifscCode']?.toString() ?? json['ifsc_code']?.toString(),
      upiId: json['upiId']?.toString() ?? json['upi_id']?.toString(),
      upiNumber: json['upiNumber']?.toString() ?? json['upi_number']?.toString(),
      platformChargePerSeat: platformChargeRaw != null
          ? (platformChargeRaw is num ? platformChargeRaw.toDouble() : double.tryParse(platformChargeRaw.toString()))
          : null,
      settlementStatus: (json['settlementStatus'] ?? json['settlement_status'] ?? 'none').toString(),
      bankDetails: json['bankDetails']?.toString() ?? json['bank_details']?.toString(),
      settlementTransactionId: json['settlementTransactionId']?.toString() ?? json['settlement_transaction_id']?.toString(),
      settlementAmount: json['settlementAmount'] != null ? double.tryParse(json['settlementAmount'].toString()) : null,
      settlementDate: json['settlementDate'] != null ? DateTime.tryParse(json['settlementDate'].toString())?.toLocal() : null,
      settlementMethod: json['settlementMethod']?.toString() ?? json['settlement_method']?.toString(),
      startedAt: json['startedAt'] != null ? DateTime.tryParse(json['startedAt'].toString())?.toLocal() : null,
      startedBy: json['startedBy']?.toString() ?? json['started_by']?.toString(),
      durationHours: json['durationHours'] != null ? double.tryParse(json['durationHours'].toString()) : null,
      expectedEndAt: json['expectedEndAt'] != null ? DateTime.tryParse(json['expectedEndAt'].toString())?.toLocal() : null,
      endedAt: json['endedAt'] != null ? DateTime.tryParse(json['endedAt'].toString())?.toLocal() : null,
      endedConfirmedBy: json['endedConfirmedBy']?.toString() ?? json['ended_confirmed_by']?.toString(),
      endedConfirmedAt: json['endedConfirmedAt'] != null ? DateTime.tryParse(json['endedConfirmedAt'].toString())?.toLocal() : null,
      adminConfirmedEndedAt: json['adminConfirmedEndedAt'] != null ? DateTime.tryParse(json['adminConfirmedEndedAt'].toString())?.toLocal() : null,
      adminConfirmedBy: json['adminConfirmedBy']?.toString() ?? json['admin_confirmed_by']?.toString(),
      settlementOverdue: json['settlementOverdue'] == true || json['settlement_overdue'] == true,
      joinedCount: int.tryParse(json['joinedCount']?.toString() ?? json['joined_count']?.toString() ?? '') ?? 0,
      paymentCount: int.tryParse(json['paymentCount']?.toString() ?? json['payment_count']?.toString() ?? '') ?? 0,
      remainingCount: int.tryParse(json['remainingCount']?.toString() ?? json['remaining_count']?.toString() ?? '') ?? 0,
      user: userMap,
      venue: venueMap,
      joiners: json['joiners'] is List ? json['joiners'] : null,
    );
  }

  /// Helper getters for lifecycle
  bool get isInProgress => status.toLowerCase() == 'in_progress';
  bool get isHostConfirmedEnded => status.toLowerCase() == 'host_confirmed_ended';
  bool get isAdminConfirmedEnded => status.toLowerCase() == 'admin_confirmed_ended';
  bool get isCompleted => status.toLowerCase() == 'completed';

  /// Remaining duration string (hh:mm:ss) until expectedEndAt
  String get remainingTimeFormatted {
    if (expectedEndAt == null) return '--:--';
    final diff = expectedEndAt!.difference(DateTime.now());
    if (diff.isNegative) return '00:00:00';
    final hours = diff.inHours.toString().padLeft(2, '0');
    final minutes = (diff.inMinutes % 60).toString().padLeft(2, '0');
    final seconds = (diff.inSeconds % 60).toString().padLeft(2, '0');
    return '$hours:$minutes:$seconds';
  }

  /// Whether the host provided bank details (either structured or legacy UPI/UPI number)
  bool get hasBankDetails =>
      (bankName != null && bankName!.isNotEmpty) ||
      (upiId != null && upiId!.isNotEmpty) ||
      (upiNumber != null && upiNumber!.isNotEmpty) ||
      (bankDetails != null && bankDetails!.isNotEmpty);

  /// Returns a display-friendly string of the bank/UPI info
  String get bankDetailsSummary {
    if (upiId != null && upiId!.isNotEmpty) return 'UPI ID: $upiId';
    if (upiNumber != null && upiNumber!.isNotEmpty) return 'UPI No: $upiNumber';
    if (bankName != null && bankName!.isNotEmpty) {
      return '$bankName · ${accountNumber ?? ''} · IFSC: ${ifscCode ?? ''}';
    }
    return bankDetails ?? 'No bank details provided';
  }

  /// Returns the dynamically confirmed participant count (joined users)
  int get actualParticipantsCount {
    if (joinedCount > 0) return joinedCount;
    if (paymentCount > 0) return paymentCount;
    if (slotsFilled > 0) return slotsFilled;
    if (joiners != null && joiners!.isNotEmpty) {
      final valid = joiners!.where((j) {
        if (j is! Map) return false;
        final st = (j['status'] ?? '').toString().toLowerCase();
        final pst = (j['paymentStatus'] ?? '').toString().toLowerCase();
        return st == 'paid' || st == 'accepted' || pst == 'paid';
      }).length;
      if (valid > 0) return valid;
    }
    return 1; // Default to host (1 person)
  }
}

File: Lunara_backend/server/src/models/StrangersMeetRequest.ts
import { Model, DataTypes, Optional } from 'sequelize';
import sequelize from '../config/database';

export enum StrangersMeetStatus {
    PENDING = 'pending',
    APPROVED = 'approved',
    START_CONFIRMATION_PENDING = 'start_confirmation_pending',
    IN_PROGRESS = 'in_progress',
    END_CONFIRMATION_PENDING = 'end_confirmation_pending',
    HOST_CONFIRMED_ENDED = 'host_confirmed_ended',
    ADMIN_CONFIRMED_ENDED = 'admin_confirmed_ended',
    SETTLED = 'settled',
    COMPLETED = 'completed',
    REJECTED = 'rejected',
    NOT_STARTED = 'not_started',
    NEEDS_HOST_CONTACT = 'needs_host_contact',
    CANCELLED = 'cancelled',
    ADMIN_RESOLVED = 'admin_resolved',
}

export enum StrangersMeetPaymentStatus {
    UNPAID = 'unpaid',
    PAID = 'paid',
}

export interface StrangersMeetRequestAttributes {
    id: string;
    userId: string;
    venueId: string;
    subject: string;
    tagline: string;
    eventDateTime: Date;
    numberOfPersons: number;
    chargesPerHead: number;          // Set by user on creation
    slotsFilled: number;             // Number of joined users
    status: StrangersMeetStatus;
    paymentAmount?: number;          // Set by admin on approval (deposit request)
    paymentStatus: StrangersMeetPaymentStatus;
    mobileNumber: string;
    alternateMobileNumber?: string;
    adminNotes?: string;
    ticketId?: string;               // Generated on payment
    ticketUrl?: string;              // Generated PDF url
    razorpayOrderId?: string;
    razorpayPaymentId?: string;
    razorpaySignature?: string;
    settlementStatus?: 'none' | 'requested' | 'approved' | 'settlement_pending' | 'paid' | 'settled';
    bankDetails?: string;         // Legacy text field (kept for backward compat)
    // v2: Structured bank payment fields (collected at creation)
    bankName?: string;
    accountNumber?: string;
    accountHolderName?: string;
    ifscCode?: string;
    upiId?: string;
    upiNumber?: string;
    platformChargePerSeat?: number; // Auto-calc'd by admin: paymentAmount / numberOfPersons
    settlementTransactionId?: string;
    settlementAmount?: number;
    settlementDate?: Date;
    settlementMethod?: string;
    foodPreference?: string;
    drinkPreference?: string;
    reminder2hSent?: boolean;
    reminder1hSent?: boolean;
    reminder30mSent?: boolean;
    // Lifecycle additions
    startedAt?: Date;
    startedBy?: string;
    durationHours?: number;
    expectedEndAt?: Date;
    endedAt?: Date;
    endedConfirmedBy?: string;
    endedConfirmedAt?: Date;
    adminConfirmedEndedAt?: Date;
    adminConfirmedBy?: string;
    settlementOverdue?: boolean;
    // Escalation & Fallback fields
    escalatedAt?: Date;
    escalationReason?: string;
    adminResolution?: string;
    adminResolutionNotes?: string;
    adminResolvedAt?: Date;
    adminResolvedBy?: string;
    hostNotStartedAt?: Date;
    hostNotStartedReason?: string;
    createdAt?: Date;
    updatedAt?: Date;
}

export interface StrangersMeetRequestCreationAttributes
    extends Optional<
        StrangersMeetRequestAttributes,
        | 'id'
        | 'status'
        | 'slotsFilled'
        | 'paymentAmount'
        | 'paymentStatus'
        | 'alternateMobileNumber'
        | 'adminNotes'
        | 'ticketId'
        | 'ticketUrl'
        | 'razorpayOrderId'
        | 'razorpayPaymentId'
        | 'razorpaySignature'
        | 'settlementStatus'
        | 'bankDetails'
        | 'bankName'
        | 'accountNumber'
        | 'accountHolderName'
        | 'ifscCode'
        | 'upiId'
        | 'upiNumber'
        | 'platformChargePerSeat'
        | 'settlementTransactionId'
        | 'settlementAmount'
        | 'settlementDate'
        | 'settlementMethod'
        | 'foodPreference'
        | 'drinkPreference'
        | 'reminder2hSent'
        | 'reminder1hSent'
        | 'reminder30mSent'
        | 'startedAt'
        | 'startedBy'
        | 'durationHours'
        | 'expectedEndAt'
        | 'endedAt'
        | 'endedConfirmedBy'
        | 'endedConfirmedAt'
        | 'adminConfirmedEndedAt'
        | 'adminConfirmedBy'
        | 'settlementOverdue'
        | 'escalatedAt'
        | 'escalationReason'
        | 'adminResolution'
        | 'adminResolutionNotes'
        | 'adminResolvedAt'
        | 'adminResolvedBy'
        | 'hostNotStartedAt'
        | 'hostNotStartedReason'
        | 'createdAt'
        | 'updatedAt'
    > { }

class StrangersMeetRequest
    extends Model<StrangersMeetRequestAttributes, StrangersMeetRequestCreationAttributes>
    implements StrangersMeetRequestAttributes {
    public id!: string;
    public userId!: string;
    public venueId!: string;
    public subject!: string;
    public tagline!: string;
    public eventDateTime!: Date;
    public numberOfPersons!: number;
    public chargesPerHead!: number;
    public slotsFilled!: number;
    public status!: StrangersMeetStatus;
    public paymentAmount?: number;
    public paymentStatus!: StrangersMeetPaymentStatus;
    public mobileNumber!: string;
    public alternateMobileNumber?: string;
    public adminNotes?: string;
    public ticketId?: string;
    public ticketUrl?: string;
    public razorpayOrderId?: string;
    public razorpayPaymentId?: string;
    public razorpaySignature?: string;
    public settlementStatus?: 'none' | 'requested' | 'approved' | 'settlement_pending' | 'paid' | 'settled';
    public bankDetails?: string;
    public bankName?: string;
    public accountNumber?: string;
    public accountHolderName?: string;
    public ifscCode?: string;
    public upiId?: string;
    public upiNumber?: string;
    public platformChargePerSeat?: number;
    public settlementTransactionId?: string;
    public settlementAmount?: number;
    public settlementDate?: Date;
    public settlementMethod?: string;
    public foodPreference?: string;
    public drinkPreference?: string;
    public reminder2hSent!: boolean;
    public reminder1hSent!: boolean;
    public reminder30mSent!: boolean;
    public startedAt?: Date;
    public startedBy?: string;
    public durationHours?: number;
    public expectedEndAt?: Date;
    public endedAt?: Date;
    public endedConfirmedBy?: string;
    public endedConfirmedAt?: Date;
    public adminConfirmedEndedAt?: Date;
    public adminConfirmedBy?: string;
    public settlementOverdue!: boolean;
    public escalatedAt?: Date;
    public escalationReason?: string;
    public adminResolution?: string;
    public adminResolutionNotes?: string;
    public adminResolvedAt?: Date;
    public adminResolvedBy?: string;
    public hostNotStartedAt?: Date;
    public hostNotStartedReason?: string;
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;
}

StrangersMeetRequest.init(
    {
        id: {
            type: DataTypes.UUID,
            defaultValue: DataTypes.UUIDV4,
            primaryKey: true,
        },
        userId: {
            type: DataTypes.UUID,
            allowNull: false,
            field: 'user_id',
            references: { model: 'users', key: 'id' },
            onDelete: 'CASCADE',
        },
        venueId: {
            type: DataTypes.UUID,
            allowNull: false,
            field: 'venue_id',
            references: { model: 'venues', key: 'id' },
            onDelete: 'CASCADE',
        },
        subject: {
            type: DataTypes.STRING(200),
            allowNull: false,
            validate: {
                len: [1, 200],
            },
        },
        tagline: {
            type: DataTypes.TEXT,
            allowNull: false,
        },
        eventDateTime: {
            type: DataTypes.DATE,
            allowNull: false,
            field: 'event_date_time',
        },
        numberOfPersons: {
            type: DataTypes.INTEGER,
            allowNull: false,
            field: 'number_of_persons',
            validate: {
                min: 21,
                max: 50,
            },
        },
        chargesPerHead: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: false,
            defaultValue: 0.0,
            field: 'charges_per_head',
        },
        slotsFilled: {
            type: DataTypes.INTEGER,
            allowNull: false,
            defaultValue: 0,
            field: 'slots_filled',
        },
        status: {
            type: DataTypes.ENUM(
                'pending',
                'approved',
                'start_confirmation_pending',
                'in_progress',
                'end_confirmation_pending',
                'host_confirmed_ended',
                'admin_confirmed_ended',
                'settled',
                'completed',
                'rejected',
                'not_started',
                'needs_host_contact',
                'cancelled',
                'admin_resolved'
            ),
            allowNull: false,
            defaultValue: 'pending',
        },
        paymentAmount: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: true,
            field: 'payment_amount',
        },
        paymentStatus: {
            type: DataTypes.ENUM('unpaid', 'paid'),
            allowNull: false,
            defaultValue: 'unpaid',
            field: 'payment_status',
        },
        mobileNumber: {
            type: DataTypes.STRING(20),
            allowNull: false,
            field: 'mobile_number',
        },
        alternateMobileNumber: {
            type: DataTypes.STRING(20),
            allowNull: true,
            field: 'alternate_mobile_number',
        },
        adminNotes: {
            type: DataTypes.TEXT,
            allowNull: true,
            field: 'admin_notes',
        },
        ticketId: {
            type: DataTypes.STRING(50),
            allowNull: true,
            field: 'ticket_id',
        },
        ticketUrl: {
            type: DataTypes.STRING(500),
            allowNull: true,
            field: 'ticket_url',
        },
        razorpayOrderId: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'razorpay_order_id',
        },
        razorpayPaymentId: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'razorpay_payment_id',
        },
        razorpaySignature: {
            type: DataTypes.STRING(255),
            allowNull: true,
            field: 'razorpay_signature',
        },
        settlementStatus: {
            type: DataTypes.ENUM('none', 'requested', 'approved', 'settlement_pending', 'paid', 'settled'),
            allowNull: true,
            defaultValue: 'none',
            field: 'settlement_status',
        },
        bankDetails: {
            type: DataTypes.TEXT,
            allowNull: true,
            field: 'bank_details',
        },
        bankName: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'bank_name',
        },
        accountNumber: {
            type: DataTypes.STRING(50),
            allowNull: true,
            field: 'account_number',
        },
        accountHolderName: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'account_holder_name',
        },
        ifscCode: {
            type: DataTypes.STRING(20),
            allowNull: true,
            field: 'ifsc_code',
        },
        upiId: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'upi_id',
        },
        upiNumber: {
            type: DataTypes.STRING(20),
            allowNull: true,
            field: 'upi_number',
        },
        platformChargePerSeat: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: true,
            defaultValue: 0,
            field: 'platform_charge_per_seat',
        },
        settlementTransactionId: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'settlement_transaction_id',
        },
        settlementAmount: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: true,
            field: 'settlement_amount',
        },
        settlementDate: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'settlement_date',
        },
        settlementMethod: {
            type: DataTypes.STRING(50),
            allowNull: true,
            field: 'settlement_method',
        },
        foodPreference: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'food_preference',
        },
        drinkPreference: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'drink_preference',
        },
        reminder2hSent: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
            field: 'reminder_2h_sent',
        },
        reminder1hSent: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
            field: 'reminder_1h_sent',
        },
        reminder30mSent: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
            field: 'reminder_30m_sent',
        },
        startedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'started_at',
        },
        startedBy: {
            type: DataTypes.UUID,
            allowNull: true,
            field: 'started_by',
        },
        durationHours: {
            type: DataTypes.FLOAT,
            allowNull: true,
            field: 'duration_hours',
        },
        expectedEndAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'expected_end_at',
        },
        endedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'ended_at',
        },
        endedConfirmedBy: {
            type: DataTypes.UUID,
            allowNull: true,
            field: 'ended_confirmed_by',
        },
        endedConfirmedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'ended_confirmed_at',
        },
        adminConfirmedEndedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'admin_confirmed_ended_at',
        },
        adminConfirmedBy: {
            type: DataTypes.UUID,
            allowNull: true,
            field: 'admin_confirmed_by',
        },
        settlementOverdue: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
            field: 'settlement_overdue',
        },
        escalatedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'escalated_at',
        },
        escalationReason: {
            type: DataTypes.TEXT,
            allowNull: true,
            field: 'escalation_reason',
        },
        adminResolution: {
            type: DataTypes.STRING(100),
            allowNull: true,
            field: 'admin_resolution',
        },
        adminResolutionNotes: {
            type: DataTypes.TEXT,
            allowNull: true,
            field: 'admin_resolution_notes',
        },
        adminResolvedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'admin_resolved_at',
        },
        adminResolvedBy: {
            type: DataTypes.UUID,
            allowNull: true,
            field: 'admin_resolved_by',
        },
        hostNotStartedAt: {
            type: DataTypes.DATE,
            allowNull: true,
            field: 'host_not_started_at',
        },
        hostNotStartedReason: {
            type: DataTypes.TEXT,
            allowNull: true,
            field: 'host_not_started_reason',
        },
    },
    {
        sequelize,
        tableName: 'strangers_meet_requests',
        underscored: true,
        timestamps: true,
        indexes: [
            { fields: ['user_id'] },
            { fields: ['venue_id'] },
            { fields: ['status'] },
            { fields: ['payment_status'] },
            { fields: ['created_at'] },
        ],
    }
);

export default StrangersMeetRequest;
